File: src/Nebula/Modules/Inventory/Models/ProductStock.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using Nebula.Common.Models;

namespace Nebula.Modules.Inventory.Models;

[BsonIgnoreExtraElements]
public class ProductStock : IGenericModel
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = string.Empty;

    /// <summary>
    /// Identificador de la empresa al que pertenece.
    /// </summary>
    public string CompanyId { get; set; } = string.Empty;

    /// <summary>
    /// Identificador Almacén.
    /// </summary>
    public string WarehouseId { get; set; } = string.Empty;

    /// <summary>
    /// Identificador Producto.
    /// </summary>
    public string ProductId { get; set; } = string.Empty;

    /// <summary>
    /// Tipo de Transacción de Inventario (Entrada o Salida).
    /// </summary>
    public string TransactionType { get; set; } = Helpers.TransactionType.ENTRADA;

    /// <summary>
    /// Cantidad de Productos.
    /// </summary>
    public decimal Quantity { get; set; } = decimal.Zero;
}

File: src/webapp/src/app/products/interfaces/product-stock.ts
export class ProductStock {
  id: any = undefined;
  companyId: string = "";
  warehouseId: string = "";
  productId: string = "";
  transactionType: string = "ENTRADA";
  quantity: number = 0;
}

export class ProductStockInfo {
  warehouseId: string = "";
  warehouseName: string = "";
  productId: string = "";
  quantity: number = 0;
}

export class ChangeQuantityStockRequestParams {
  constructor(
    public warehouseId: string = "",
    public productId: string = "",
    public productLoteId: string = "-",
    public quantity: number = 0) {
  }
}
